libs/s25main/ingameWindows/iwTempleBuilding.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "iwTempleBuilding.h"
+#include "WineLoader.h"
+#include "buildings/nobTemple.h"
+#include "controls/ctrlImage.h"
+#include "controls/ctrlImageButton.h"
+#include "factories/GameCommandFactory.h"
+#include "world/GameWorldView.h"
+
+iwTempleBuilding::iwTempleBuilding(GameWorldView& gwv, GameCommandFactory& gcFactory, nobUsual* const building)
+    : iwBuilding(gwv, gcFactory, building, Extent(226, 223))
+{
+    GetCtrl<Window>(1)->SetPos(DrawPoint(117, 160));
+    GetCtrl<ctrlImage>(3)->SetImage(
+      wineaddon::GetTempleProductionModeTex(static_cast<nobTemple*>(building)->GetProductionMode()));
+    AddImageButton(8, DrawPoint(130, 176), Extent(34, 32), TextureColor::Grey,
+                   wineaddon::GetTempleProductionModeTex(static_cast<nobTemple*>(building)->GetProductionMode()));
```

Comment:
Do you really want an image and the button with the same image? If so I'd guess you need to update both of them on click

Or can the button maybe show the next mode? I.e. click the button to switch to that shown mode and the image shows the current mode

Reply:
This was a copy and past error. The first image should never change and is fixed.

Reply:
If you like I can add a larger output icon for the top right of the window: the gold, iron ore, coal and stone. Then use the smaller one for the button and I can add smaller icons for iron ore, coal and stone for the button toggle at the bottom?